lib/kube/kubeconfig/kubeconfig.go

```diff
@@ -359,9 +367,52 @@ func SelectContext(teleportCluster, kubeCluster string) error {
 	if _, ok := kc.Contexts[kubeContext]; !ok {
 		return trace.NotFound("kubeconfig context %q not found", kubeContext)
 	}
+	setSelectedExtension(kc.Contexts, kc.CurrentContext, teleportCluster)
 	kc.CurrentContext = kubeContext
 	if err := Save("", *kc); err != nil {
 		return trace.Wrap(err)
 	}
 	return nil
 }
+
+const selectedExtension = "teleport-prev-selec-ctx"
+
+// setSelectedExtension sets an extension to indentify that the current non-teleport
+// context was selected before introducing Teleport contexts in kubeconfig.
+// If the currentContext is not from Teleport, this function adds the following
+// extensions:
+//   - extension: null
+//     name: teleport-prev-selec-ctx
+//
+// Only one context is allowed to have the selected extension. If other context has it,
+// this function deletes it and introduces it in the desired context.
+func setSelectedExtension(contexts map[string]*clientcmdapi.Context, prevCluster string, teleportCluster string) {
```

Comment:
Wonder why the top [config.extensions](https://github.com/kubernetes/client-go/blob/674e53a7cb8f1ea160ac4485ffc7582e1a790215/tools/clientcmd/api/v1/types.go#L51) was not used  but I'm fine with relaying on `config.contexts.extensions`

Reply:
I decided to use context extensions because when merging multiple kubeconfig files, `config.extensions` might be overwritten/merged which is difficult to resolve.